test/AutoRest.TestServer.Tests/paging.cs

```diff
@@ -0,0 +1,375 @@
+
+        [Test]
+        [Ignore("Needs long running operation logic: https://github.com/Azure/autorest.csharp/issues/399")]
+        public Task PagingMultipleLRO() => TestStatus(async (host, pipeline) => { await Task.FromException(new Exception()); return null; });
+
+        [Test]
+        public Task PagingMultiplePath() => Test(async (host, pipeline) =>
+        {
+            var id = 1;
+            var pageNumber = 1;
+            var product = "Product";
+            var offset = 42;
+            var linkPart = $"/paging/multiple/withpath/page/{offset}/";
+            var result = await new PagingOperations(ClientDiagnostics, pipeline, host).GetMultiplePagesWithOffsetAsync(null, null, offset, null);
+            var resultPage = Page.FromValues(result.Value.Values, result.Value.NextLink, result.GetRawResponse());
+            while (resultPage.ContinuationToken != null)
+            {
+                Assert.AreEqual(id, resultPage.Values.First().Properties.Id);
+                Assert.AreEqual(product, resultPage.Values.First().Properties.Name);
+                StringAssert.EndsWith($"{linkPart}{++pageNumber}", resultPage.ContinuationToken);
+                result = await new PagingOperations(ClientDiagnostics, pipeline, host).GetMultiplePagesWithOffsetNextPageAsync(null, null, null, resultPage.ContinuationToken);
+                resultPage = Page.FromValues(result.Value.Values, result.Value.NextLink, result.GetRawResponse());
+                if (product == "Product")
+                {
+                    id += offset;
+                    product = "product";
+                }
+
+                id++;
+            }
+            Assert.AreEqual(10, pageNumber);
+            Assert.AreEqual(id, resultPage.Values.First().Properties.Id);
+            Assert.AreEqual(product, resultPage.Values.First().Properties.Name);
+
+            id = 1;
+            pageNumber = 1;
+            product = "Product";
+            var pageable = new PagingOperations(ClientDiagnostics, pipeline, host).GetMultiplePagesWithOffsetPageableAsync(null, null, offset, null);
+            await foreach (var page in pageable.AsPages())
+            {
+                Assert.AreEqual(id, page.Values.First().Properties.Id);
+                Assert.AreEqual(product, page.Values.First().Properties.Name);
+                if (pageNumber == 10)
+                {
+                    Assert.IsNull(page.ContinuationToken);
+                }
+                else
+                {
+                    StringAssert.EndsWith($"{linkPart}{++pageNumber}", page.ContinuationToken);
+                }
+                if (product == "Product")
+                {
+                    id += offset;
+                    product = "product";
+                }
+
+                id++;
+            }
+            Assert.AreEqual(10, pageNumber);
+        }, true);
+
+        [Test]
+        [Ignore("Needs retry logic: https://github.com/Azure/autorest.csharp/issues/398")]
```

Comment:
We have retry logic in the pipeline by default. This should work.

Reply:
As it stands, we don't have multiple response types implemented. So, the 500 response code won't be there for this to work, right? So, I shelved this until we have things set up. And, I don't know how retries work when doing the IEnumerable/IAsyncEnumerable stuff (haven't designed it that far yet). It is okay for this PR to not do everything related to paging. It is an initial paging implementation.